main/main.go

```diff
@@ -39,159 +45,204 @@ const (
 	seqNumberSetError            // 6
 	statusReportingError         // 7
 	settingsNotFoundError        // 8
+	versionMismatchError         // 9
+	jsonParsingError             // 10
 )
 
-func install() {
-	infoLogger.Println("Installed Succesfully.")
-}
-
-func enable() {
-	infoLogger.Println("Enabled Successfully.")
-}
-
-func disable() {
-	infoLogger.Println("Disabled Successfully.")
+// extension specific PublicSettings
+type extensionPublicSettings struct {
+	Name string `json:"name"`
 }
 
-func uninstall() {
-	infoLogger.Println("Uninstalled Successfully.")
+// extension specific ProtectedSettings
+type extensionPrivateSettings struct {
+	SecretString       string   `json:"secretString"`
+	SecretScript       string   `json:"secretScript"`
+	FileURLs           []string `json:"fileUris"`
+	StorageAccountName string   `json:"storageAccountName"`
```

Comment:
You dont need most of this extra stuff, just keep `SecretString` and remove the rest